### DISCUSSION TITLE
Contrast for UI Elements (Non Text)

### DISCUSSION THREAD

**Discussion by Garz** [Myndex/SAPC-APCA#124] [Q&A]
Hello! - looking for some guidance around non text requirements such as buttons, cards, inputs, etc.

The documentation for text, icons and even decorative aspects like dividers is pretty straight forward to me, however where 3.0/APCA and even 2.0 are not as clear... is providing guidance on non text contrast levels as it pertains to really common UI elements and components.

Take a button for example: Ignore the text and just focus on the button fill in contrast to the background it sits on. What is very common to see is an alpha value of white or an off-white say 8% on top of black or an off-black.

If you were to test this contrast - it would fail both methods for both states of the button fill.

Another example is GitHub it's self, if you hover over the top tabs and see the hover container fill.

I guess my question is, does this fall under the 15c req, 30c req, or does it not need to pass at all?

Appreciate the help!

![Screenshot 2024-08-21 at 3 07 27 AM](https://github.com/user-attachments/assets/23f80459-0f1e-4c8a-8c2c-b6a80e6203d3)

**Comment by kallypso-starmoss** [comment]
Would love to hear more thoughts on this as well!
